### PR TITLE
Change video buffer object upload logic; Need to get line witdth per plane and t…

### PR DESCRIPTION
**Pull request by js0701**
…he actual height of the plane, and upload line by line with strides

### CODE REVIEW

File: tests/common/videolayerrenderer.cpp [hunk 1]
```diff
@@ -45,6 +46,155 @@ bool VideoLayerRenderer::Init(uint32_t width, uint32_t height, uint32_t format,
   return true;
 }
 
+static int get_bpp_from_format(uint32_t format, size_t plane)
+{
+	//assert(plane < drv_num_planes_from_format(format));
+	switch (format) {
+	  case DRM_FORMAT_BGR233:
+	  case DRM_FORMAT_C8:
```
Comment (kalyankondapally) [inline tests/common/videolayerrenderer.cpp]: Fix indentation, git diff | clang-format-diff-3.5 -p 1 -style=file  should do it before commiting the changes

File: tests/common/videolayerrenderer.cpp [hunk 2]
```diff
@@ -45,6 +46,155 @@ bool VideoLayerRenderer::Init(uint32_t width, uint32_t height, uint32_t format,
...
+}
+
+static uint32_t get_linewidth_from_format(uint32_t format, uint32_t width, size_t plane)
+{
+
+   int stride = width * ((get_bpp_from_format(format, plane) + 7) / 8);
+
+	/*
+	 * Only downsample for certain multiplanar formats which have horizontal
+	 * subsampling for chroma planes.  Only formats supported by our drivers
```
Comment (kalyankondapally) [inline tests/common/videolayerrenderer.cpp]: Same as above fix indentation

File: tests/common/videolayerrenderer.cpp [hunk 3]
```diff
@@ -45,6 +46,155 @@ bool VideoLayerRenderer::Init(uint32_t width, uint32_t height, uint32_t format,
...
+	  case DRM_FORMAT_RGBX1010102:
+	  case DRM_FORMAT_RGBX8888:
+	  case DRM_FORMAT_XBGR2101010:
+	  case DRM_FORMAT_XBGR8888:
+	  case DRM_FORMAT_XRGB2101010:
+	  case DRM_FORMAT_XRGB8888:
+		return 32;
+	}
+
+	fprintf(stderr, "drv: UNKNOWN FORMAT %d\n", format);
```
Comment (kalyankondapally) [inline tests/common/videolayerrenderer.cpp]: Can we just ETRACE here ? include platform defines header.

Comment (js0701) [inline tests/common/videolayerrenderer.cpp]: OK. I think ETRACE is better

File: tests/common/gllayerrenderer.cpp [hunk 4]
```diff
@@ -81,6 +81,8 @@ void GLLayerRenderer::Draw(int64_t* pfence) {
 
   glDrawFrame();
 
+  int64_t gpu_fence_fd = -1;
```
Comment (kalyankondapally) [inline tests/common/gllayerrenderer.cpp]: Hmm I actually think we should get of this logic since HWC should provide the synchronization.

**Comment by kalyankondapally** [comment]
Can you separate tests/common/gllayerrenderer.cpp to a separate pull request. Rest merged.

**Pull request closed without merging**